stock_picking_backorder_to_sale/views/stock_view.xml

```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright 2017 Carlos Dauden <carlos.dauden@tecnativa.com>
+     License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<openerp>
+<data>
+
+<record id="view_picking_form" model="ir.ui.view">
+    <field name="model">stock.picking</field>
+    <field name="inherit_id" ref="stock.view_picking_form"/>
+    <field name="arch" type="xml">
+        <xpath expr="//button[@name='do_unreserve']" position="after">
+            <button name="button_create_sale" string="Create Sale" type="object" groups="base.group_user"
+                    attrs="{'invisible': [('state', '=', 'done')]}"/>
```

Comment:
I don't know if it have too much sense to be able to use this from an internal transfer or a receipt, it may lead to wrong workflow.